services/server/getEvents.ts

```diff
@@ -0,0 +1,116 @@
+import client from "@/tina/client";
+
+export const getPastEvents = async (top, presenterName) => {
+  console.log("presenterName", presenterName);
+  const eventClient = await client.queries.getPastEventsQuery(
+    formatEventParams(top, presenterName)
+  );
+  return await fetchEventsWithClient(eventClient, presenterName, top);
+};
+
+export const getUpcomingEvents = async (top, presenterName) => {
+  const eventClient = await client.queries.getFutureEventsQuery(
+    formatEventParams(top, presenterName)
+  );
+  return await fetchEventsWithClient(eventClient, presenterName, top);
+};
+
+const formatEvent = (event) => {
+  const url = event.url && fixRelativeUrl(event.url);
+  return {
+    Presenter: event.presenterName,
+    Url: {
+      Description: event.title, // this is alt text for the link, ask matt what he wants to do for it
+      Url: url,
+    },
+    title: event.title,
+    Thumbnail: {
+      Url: fixRelativeUrl(event.thumbnail),
+      Description: event.thumbnailDescription,
+    },
+    PresenterProfileUrl:
+      event.presenterProfileUrl && fixRelativeUrl(event.presenterProfileUrl),
+    StartDateTime: event.startDateTime,
+    EndDateTime: event.endDateTime,
+    StartShowBannerDateTime: event.startShowBannerDateTime,
+    EndShowBannerDateTime: event.endShowBannerDateTime,
+    LiveStreamUrl: event.youTubeId
+      ? `https://www.youtube.com/watch?v=${event.youTubeId}`
+      : null,
+    CalendarType: event.calendarType,
+    City: event.city,
+    Enabled: event.enabled,
+    EventShortDescription: descriptionToHTML(event.description),
+    ShowOnPowerSessions: "No",
+    NoteInternal: event.internalNote,
+    HasVideo: event.youTubeId ? "Yes" : "No",
+    YouTubeId: event.youTubeId,
+    PresenterDescription:
+      event.presenterList && event.presenterListlength > 0
+        ? event.presenterList[0].presenter.about
+        : null,
+  };
+};
+export const fetchEventsWithClient = async (
+  eventClient,
+  presenterName: string | undefined,
+  top
+) => {
+  if (top) {
+    top = parseInt(top);
+  }
+  const events = [];
+  // TODO: remove client side after fixing events with multiple presenters in the name
+  for (const event of eventClient.data.eventsCalendarConnection.edges) {
+    if (
+      !presenterName ||
+      (event.node.presenterName &&
+        event.node.presenterName
+          .toLowerCase()
+          .includes(presenterName.toLowerCase()))
+    ) {
+      events.push(formatEvent(event.node));
+    }
+
+    if (events.length === (top || 10)) {
+      break;
+    }
+  }
+  return events;
+};
+
+const fixRelativeUrl = (url: string) => {
+  if (url.startsWith("/")) {
+    return `https://ssw.com.au${url}`;
+  }
+  return url;
+};
+
+const descriptionToHTML = (description) => {
```

Comment:
couldn't this be fixed at the source?

Reply:
I asked Jack P over teams and he told me it's not unfortunately.

Reply:
As per our convo, check if people actually uses this, if not just remove it.

Reply:
Done - I couldn't see it being used in the code on SSW People so I removed it

**Note**: Even though it's not being used anywhere in the code on people the service still expects this prop. I'll update the service after this.

Reply:
Done - I couldn't see it being used in the code on SSW People so I removed it

Note: Even though it's not being used anywhere in the code on people the service still expects this prop. I'll update the service after this.